### PR TITLE
Feature/em 2841 convert and display new document types

### PR DESCRIPTION
**Pull request by jackmunkonge**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EM-2841
https://tools.hmcts.net/jira/browse/EM-2842
https://tools.hmcts.net/jira/browse/EM-2843
https://tools.hmcts.net/jira/browse/EM-2844
https://tools.hmcts.net/jira/browse/EM-2845

### Change description ###
convert and display new document types

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/media-viewer/src/lib/store/reducers/document.reducer.ts [hunk 1]
```diff
@@ -1,10 +1,15 @@
 import * as fromActions from '../actions/document.action';
 
 export interface DocumentState {
+  convertedDocumentInfo: {
```
Comment (meganaslattery) [inline projects/media-viewer/src/lib/store/reducers/document.reducer.ts:4]: Maybe just call it convertedDocument?

File: projects/media-viewer/src/lib/store/effects/document.effects.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    ofType(documentActions.CONVERT),
+    map((action: documentActions.Convert) => action.payload),
+    exhaustMap((docConversionPayload) => {
+      return this.documentConversionService.convert(docConversionPayload).pipe(
+        map((result: HttpResponse<Blob>) => {
+          const url = URL.createObjectURL(result.body);
+          return new documentActions.ConvertSuccess(url);
+        }),
+        catchError(error => {
+          return of(new documentActions.ConvertFail(error));
```
Comment (benouaer) [inline projects/media-viewer/src/lib/store/effects/document.effects.ts:27]: do we need the of() here, does it work without it?

File: projects/media-viewer/src/lib/store/reducers/document.reducer.ts [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
+    url: string;
+    error: string;
+  };
   documentId: string;
   pages: DocumentPages;
   loaded: boolean;
   loading: boolean;
+  error: string;
```
Comment (benouaer) [inline projects/media-viewer/src/lib/store/reducers/document.reducer.ts:12]: what's the error here for?

File: projects/media-viewer/src/lib/viewers/convertible-content-viewer/convertible-content-viewer.component.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  @Input() enableAnnotations: boolean;
+  @Input() enableRedactions: boolean;
+  @Input() annotationSet: AnnotationSet | null;
+
+
+  @Output() mediaLoadStatus = new EventEmitter<ResponseType>();
+  @Output() viewerException = new EventEmitter<ViewerException>();
+  @Output() documentTitle = new EventEmitter<string>();
+
+  private $subscription: Subscription = new Subscription();
```
Comment (benouaer) [inline projects/media-viewer/src/lib/viewers/convertible-content-viewer/convertible-content-viewer.component.ts:32]: you don't instantiate the subscription this way, you just set it to the first subscription you create

File: projects/media-viewer/src/lib/viewers/convertible-content-viewer/convertible-content-viewer.component.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  convertedUrl: string;
+
+  constructor(private store: Store<fromStore.DocumentState>) {
+  }
+
+  ngOnInit(): void {
+    this.store.dispatch(new fromDocumentActions.Convert(this.extractDMStoreDocId(this.originalUrl)));
+    this.$subscription.add(this.store.pipe(select(fromSelectors.getConvertedDocumentInfo), filter(value => !!value))
```
Comment (benouaer) [inline projects/media-viewer/src/lib/viewers/convertible-content-viewer/convertible-content-viewer.component.ts:40]: as per subscription comment above, here you would set the subscription rather than add to it

File: projects/media-viewer/src/lib/viewers/convertible-content-viewer/convertible-content-viewer.component.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  selector: 'mv-conversion-viewer',
+  templateUrl: './convertible-content-viewer.component.html',
+  encapsulation: ViewEncapsulation.None
+})
+export class ConvertibleContentViewerComponent implements OnInit, OnDestroy {
+
+  @Input() originalUrl;
+  @Input() downloadFileName: string;
+
+  @Input() height = 'calc(100vh - 32px)';
```
Comment (benouaer) [inline projects/media-viewer/src/lib/viewers/convertible-content-viewer/convertible-content-viewer.component.ts:21]: this is already defaulted in media-viewer, so there's no need to default it here

File: projects/media-viewer/src/lib/viewers/convertible-content-viewer/document-conversion-api.service.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpResponse } from '@angular/common/http';
+import { Observable, of } from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+
+@Injectable()
+export class DocumentConversionApiService {
+
+  public documentConversionUrl = '/doc-assembly/convert';
```
Comment (benouaer) [inline projects/media-viewer/src/lib/viewers/convertible-content-viewer/document-conversion-api.service.ts:9]: why is this public? it should be a constant as well

File: src/app/media-viewer-wrapper/media-viewer-wrapper.component.ts [hunk 8]
```diff
@@ -41,11 +41,20 @@ export class MediaViewerWrapperComponent implements AfterContentInit {
       this.setDocumentUrl(this.pdfUrl);
     } else if (newTab === 'image') {
       this.setDocumentUrl(this.imageUrl);
-    } else {
+    } else if (newTab === null){
```
Comment (benouaer) [inline src/app/media-viewer-wrapper/media-viewer-wrapper.component.ts:44]: is this change needed?

File: projects/media-viewer/src/lib/store/effects/document.effects.ts (hunk 2 above)
Comment (jackmunkonge) [reply projects/media-viewer/src/lib/store/effects/document.effects.ts:27]: Yes, it doesn't compile without it

File: projects/media-viewer/src/lib/store/reducers/document.reducer.ts (hunk 1 above)
Comment (jackmunkonge) [reply projects/media-viewer/src/lib/store/reducers/document.reducer.ts:4]: Amended

File: projects/media-viewer/src/lib/store/reducers/document.reducer.ts (hunk 3 above)
Comment (jackmunkonge) [reply projects/media-viewer/src/lib/store/reducers/document.reducer.ts:12]: No idea, removed

File: projects/media-viewer/src/lib/viewers/convertible-content-viewer/convertible-content-viewer.component.ts (hunk 6 above)
Comment (jackmunkonge) [reply projects/media-viewer/src/lib/viewers/convertible-content-viewer/convertible-content-viewer.component.ts:21]: amended

File: projects/media-viewer/src/lib/viewers/convertible-content-viewer/convertible-content-viewer.component.ts (hunk 5 above)
Comment (jackmunkonge) [reply projects/media-viewer/src/lib/viewers/convertible-content-viewer/convertible-content-viewer.component.ts:40]: amended

File: projects/media-viewer/src/lib/viewers/convertible-content-viewer/document-conversion-api.service.ts (hunk 7 above)
Comment (jackmunkonge) [reply projects/media-viewer/src/lib/viewers/convertible-content-viewer/document-conversion-api.service.ts:9]: It says class member cannot be const, so just changed to private

File: src/app/media-viewer-wrapper/media-viewer-wrapper.component.ts (hunk 8 above)
Comment (jackmunkonge) [reply src/app/media-viewer-wrapper/media-viewer-wrapper.component.ts:44]: Not anymore, amended

File: projects/media-viewer/src/lib/viewers/convertible-content-viewer/document-conversion-api.service.ts (hunk 7 above)
Comment (benouaer) [reply projects/media-viewer/src/lib/viewers/convertible-content-viewer/document-conversion-api.service.ts]: maybe set it as readonly as well
